clients/java/src/main/java/io/camunda/zeebe/client/impl/command/FailJobCommandImpl.java

```diff
@@ -32,78 +35,137 @@
 import java.time.Duration;
 import java.util.concurrent.TimeUnit;
 import java.util.function.Predicate;
+import org.apache.hc.client5.http.config.RequestConfig;
 
 public final class FailJobCommandImpl extends CommandWithVariables<FailJobCommandStep2>
     implements FailJobCommandStep1, FailJobCommandStep2 {
 
   private final GatewayStub asyncStub;
-  private final Builder builder;
+  private final Builder grpcRequestObjectBuilder;
   private final Predicate<StatusCode> retryPredicate;
   private Duration requestTimeout;
+  private boolean useRest;
+  private final JobFailRequest httpRequestObject;
+  private final HttpClient httpClient;
+  private final RequestConfig.Builder httpRequestConfig;
+  private final long jobKey;
+  private final JsonMapper jsonMapper;
 
   public FailJobCommandImpl(
       final GatewayStub asyncStub,
       final JsonMapper jsonMapper,
       final long key,
       final Duration requestTimeout,
-      final Predicate<StatusCode> retryPredicate) {
+      final Predicate<StatusCode> retryPredicate,
+      final HttpClient httpClient,
+      final boolean preferRestOverGrpc) {
     super(jsonMapper);
     this.asyncStub = asyncStub;
     this.requestTimeout = requestTimeout;
     this.retryPredicate = retryPredicate;
-    builder = FailJobRequest.newBuilder();
-    builder.setJobKey(key);
+    grpcRequestObjectBuilder = FailJobRequest.newBuilder();
+    grpcRequestObjectBuilder.setJobKey(key);
+    this.httpClient = httpClient;
+    httpRequestConfig = httpClient.newRequestConfig();
+    httpRequestObject = new JobFailRequest();
+    useRest = preferRestOverGrpc;
+    jobKey = key;
+    this.jsonMapper = jsonMapper;
```

Comment:
🔧 Why do we need a second property for the `jsonMapper`? The base class already provides it through the `objectMapper` property.

Reply:
Good catch 👍